File: app/src/main/java/com/example/mapsassignment/ResultsActivity.java
package com.example.mapsassignment;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;

import com.google.android.material.bottomnavigation.BottomNavigationView;


public class ResultsActivity extends AppCompatActivity {

    private final String TAG = this.getClass().getSimpleName();
    private String location;

    private Fragment selectedFragment;
    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.restaurants:
                    setLocation();
                    selectedFragment = new MasterFragment("restaurants", location);
                    getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                            selectedFragment).commit();
                    return true;
                case R.id.fuel:
                    setLocation();
                    selectedFragment = new MasterFragment("fuel", location);
                    getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                            selectedFragment).commit();
                    return true;
            }
            return false;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_results);
        BottomNavigationView navView = findViewById(R.id.nav_view);
        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);

        setLocation();

        //Set the default fragment to be Restaurant
        selectedFragment = new MasterFragment("restaurants", location);
        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                selectedFragment).commit();

        //Hide status bar and actionbar

        Toolbar toolbar = findViewById(R.id.toolbar);
//        setSupportActionBar(toolbar);
//        getSupportActionBar().hide();
    }

    private void setLocation() {
        Intent intent = getIntent();
        location = intent.getStringExtra("location");
    }
}

File: app/src/main/java/com/example/mapsassignment/ResultsActivity.kt
package com.example.mapsassignment

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class ResultsActivity : AppCompatActivity() {
    private val TAG = this.javaClass.simpleName
    private var location: String? = null
    private var selectedFragment: Fragment? = null
    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.restaurants -> {
                setLocation()
                selectedFragment = MasterFragment("restaurants", location)
                supportFragmentManager.beginTransaction().replace(R.id.fragment_container,
                        selectedFragment!!).commit()
                return@OnNavigationItemSelectedListener true
            }
            R.id.fuel -> {
                setLocation()
                selectedFragment = MasterFragment("fuel", location)
                supportFragmentManager.beginTransaction().replace(R.id.fragment_container,
                        selectedFragment!!).commit()
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_results)
        val navView = findViewById<BottomNavigationView>(R.id.nav_view)
        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
        setLocation()

        //Set the default fragment to be Restaurant
        selectedFragment = MasterFragment("restaurants", location)
        supportFragmentManager.beginTransaction().replace(R.id.fragment_container,
                selectedFragment!!).commit()
    }

    private fun setLocation() {

        val intent = intent
        location = intent.getStringExtra("location")
    }
}